src/autosuggest/load-more-controller.ts

```diff
@@ -36,7 +36,12 @@ export const useAutosuggestLoadMore = ({
     samePage: boolean;
     filteringText?: string;
   }) => {
-    if (filteringText === undefined || lastFilteringText.current !== filteringText) {
+    if (
+      filteringText === undefined ||
+      lastFilteringText.current !== filteringText ||
+      // special handling for clear input click
+      (filteringText === '' && lastFilteringText.current === '')
```

Comment:
This looks very hacky. We have fireLoadMoreOnScroll, fireLoadMoreOnRecoveryClick, fireLoadMoreOnInputFocus, fireLoadMoreOnInputChange. Would it make sense to introduce one for clear button click specifically?

Reply:
I agree, and I thought about that. The problem is, `onFocus` will be triggered before on click, and it fires `onLoadItems` with `currentFilteringText` being non-empty. I couldn't find a good way, other than maybe start checking click target, but that sounds like a overcomplicating things